src/nvim/ui.c

```diff
@@ -536,7 +536,21 @@ void ui_flush(void)
   if (!ui_active()) {
     return;
   }
+
+  static bool was_busy = false;
+
   cmdline_ui_flush();
+
+  if (State != MODE_CMDLINE && curwin->w_floating) {
+    if (curwin->w_config.hide) {
+      ui_call_busy_start();
+      was_busy = true;
+    }
+  } else if (was_busy) {
```

Comment:
The branching here doesn't seem correct. When switching from a hidden floating window to a non-hidden floating window, the cursor isn't remade visible.